Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -485,11 +485,33 @@ void WebPageProxy::ProcessActivityState::takeCapturingActivity()
     m_isCapturingActivity = m_page.process().throttler().foregroundActivity("View is capturing media"_s).moveToUniquePtr();
 }
 
+#if PLATFORM(MAC)
+// On macOS, we opt out pages out of process suspension if they used the Notification API to avoid breaking
+// some use cases.
+// In particular, we opt out if either:
+// - The page showed a notification
+// - The page called navigator.permissions.query({ name: "notifications" }) and it returned true.
+// - The page accessed Notification.permission and it returned "granted".
+// - The page requested permission via Notification.requestPermission() and it was granted.
+// This gets reset whenever a new main frame load commits inside the page.
+void WebPageProxy::ProcessActivityState::takeLikelyToUseNotificationsActivity()
+{
+    if (!m_likelyToUseNotificationsActivity)
+        m_likelyToUseNotificationsActivity = m_page.process().throttler().backgroundActivity("View is likely to use notifications"_s).moveToUniquePtr();
```

Comment:
If this results in `ProcessThrottler` taking a `WebKit:Background` assertion this may be an issue. `WebKit:Background` is invalidated by RunningBoard after 30 seconds. On macOS we should probably use `WebKit:IndefiniteBackground` which was recently added by RunningBoard

Reply:
Oh, that's good to know. Our background assertion on macOS should really be a IndefiniteBackground it seems.

Reply:
Hopefully addressed via https://github.com/WebKit/WebKit/pull/13278/files#r1180784260.